src/main/java/com/bookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -41,24 +38,29 @@ protected ResponseEntity<Object> handleHttpRequestMethodNotSupported(
             HttpHeaders headers,
             HttpStatusCode status,
             WebRequest request) {
-        Map<String, Object> body = new LinkedHashMap<>();
-        body.put("timestamp", LocalDateTime.now());
-        body.put("status", HttpStatus.METHOD_NOT_ALLOWED);
-        body.put("your request", request);
-        body.put("correct request", "/api/books/id");
-        return new ResponseEntity<>(body, headers, status);
+        ResponseBody responseBody = new ResponseBody(LocalDateTime.now(),
+                HttpStatus.METHOD_NOT_ALLOWED, List.of(ex.getMessage()));
+        return new ResponseEntity<>(responseBody, headers, status);
     }
 
     @ExceptionHandler(EntityNotFoundException.class)
     public ResponseEntity<Object> handleEntityNotFoundException(
             EntityNotFoundException ex
     ) {
-        Map<String, Object> body = new LinkedHashMap<>();
-        body.put("timestamp", LocalDateTime.now());
-        body.put("status", HttpStatus.NOT_FOUND);
-        body.put("error", ex.getMessage());
+        ResponseBody responseBody = new ResponseBody(LocalDateTime.now(),
+                HttpStatus.NOT_FOUND, List.of(ex.getMessage()));
         int exceptionStatus = 404;
-        return new ResponseEntity<>(body, HttpStatusCode.valueOf(exceptionStatus));
+        return new ResponseEntity<>(responseBody, HttpStatusCode.valueOf(exceptionStatus));
+    }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<Object> handleRegistrationException(
+            RegistrationException ex
+    ) {
+        ResponseBody responseBody = new ResponseBody(LocalDateTime.now(),
+                HttpStatus.BAD_REQUEST, List.of(ex.getMessage()));
+        int exceptionStatus = 400;
+        return new ResponseEntity<>(responseBody, HttpStatusCode.valueOf(exceptionStatus));
```

Comment:
better to not hardcode the status code. You can use `HttpStatus.CODE_NAME.value()`
```suggestion
        return new ResponseEntity<>(responseBody, HttpStatus.CODE_NAME.value());
```